bundles/binding/org.openhab.binding.buderus/src/main/java/org/openhab/binding/buderus/internal/BuderusBinding.java

```diff
@@ -0,0 +1,354 @@
+
+			String stxafterNakString = (String)config.get("STXafterNAK");
+			if (StringUtils.isNotBlank(stxafterNakString) && stxafterNakString.equals("true")) {
+				J3964R.STXafterNAK = true;
+			}
+			else if (StringUtils.isNotBlank(stxafterNakString) && stxafterNakString.equals("false")) {
+				J3964R.STXafterNAK = false;
+			}
+			logger.debug("J3964R.STXafterNAK: " + J3964R.STXafterNAK);
+
+			
+			if(port != null && initialize(port,baud))
+				setProperlyConfigured(true);	
+			else{
+				logger.error("Could not initialize connection to gateway on serial port " + port);
+				throw new ConfigurationException(port, "Could not initialize connection to gateway on serial port");
+			}
+		}
+	}
+
+	/**
+	 * Initialize the 3964r procedure for the connection to the gateway via serial port
+	 * 
+	 * @param port serial port to the gateway
+	 * @param baud baud rate for the connection
+	 * @return success state of the initialization
+	 */
+	@SuppressWarnings("rawtypes")
+	public boolean initialize(String port, int baud) {
+		boolean success;
+		// parse ports and if the default port is found, initialized the reader
+		CommPortIdentifier portId = null;
+		Enumeration portList = CommPortIdentifier.getPortIdentifiers();
+		while (portList.hasMoreElements()) {
+			CommPortIdentifier id = (CommPortIdentifier) portList.nextElement();
+			if (id.getPortType() == CommPortIdentifier.PORT_SERIAL) {
+				if (id.getName().equals(port)) {
+					logger.debug("Serial port " + port + " has been found");
+					portId = id;
+				}
+			}
+		}
+		if (portId != null) {
+			// initialize 3964R procedure via serial port
+			try {
+				proc3964R = new J3964R(port, baud, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE); 
+				//Thread starts automatically, no start() necessary
+				//do not use the default constructor due to changed PARITY!
+			} catch (PortInUseException e) {
+				logger.error("PortInUseException" + e.getMessage());
+			} 
+			logger.debug("Procedure 3964R initailized at port " + port);     
+
+			//Set speed correctly
+			int mySpeed = (baud / 2400) * 2400;
+			if (mySpeed <= 2400){
+				mySpeed = 2400;
+			}
+			try {
+				proc3964R.setSpeed(mySpeed);
+			} catch (UnsupportedCommOperationException e) {
+				logger.error("UnsupportedCommOperationException" + e.getMessage());
+			}
+			proc3964R.addJ3964RListener(this);
+			success = true;
+		} else {
+			StringBuilder sb = new StringBuilder();
+			portList = CommPortIdentifier.getPortIdentifiers();
+			while (portList.hasMoreElements()) {
+				CommPortIdentifier id = (CommPortIdentifier) portList.nextElement();
+				if (id.getPortType() == CommPortIdentifier.PORT_SERIAL) {
+					sb.append(id.getName() + "\n");
+				}
+			}
+			logger.error("Serial port " + port + " could not be found. Available ports are:\n-----" + sb.toString() + "\n-----");
+			success = false;
+		}
+		return success;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void dataReceived(J3964REvent e) {
```

Comment:
You should consider inverting the if conditionals in this method to reduce nesting. Not required, but makes for cleaner code.